bigquery-flash-data-sync/internal/config/config.go

```diff
@@ -0,0 +1,319 @@
+	"strings"
+	"time"
+
+	"github.com/wso2-open-operations/common-tools/bigquery-flash-data-sync/internal/model"
+	"go.uber.org/zap"
+)
+
+const (
+	DatabasesKey = "SYNC_DATABASES"
+
+	DefaultDBHost     = "DB_HOST"
+	DefaultDBPort     = "DB_PORT"
+	DefaultDBType     = "DB_TYPE"
+	DBMaxOpenConns    = "DB_MAX_OPEN_CONNECTIONS"
+	DBMaxIdleConns    = "DB_MAX_IDLE_CONNECTIONS"
+	DBConnMaxLifetime = "DB_CONN_MAX_LIFETIME"
+
+	GCPProjectID = "GCP_PROJECT_ID"
+	BQDatasetID  = "BQ_DATASET_ID"
+
+	SyncTimeoutKey      = "SYNC_TIMEOUT"
+	DateFormatKey       = "DATE_FORMAT"
+	DefaultBatchSizeKey = "DEFAULT_BATCH_SIZE"
+
+	DryRunKey              = "DRY_RUN"
+	CreateTablesKey        = "AUTO_CREATE_TABLES"
+	TruncateOnSyncKey      = "TRUNCATE_ON_SYNC"
+	MaxRowParseFailuresKey = "MAX_ROW_PARSE_FAILURES"
+)
+
+// LoadConfig reads all required environment variables and builds database connection strings.
+// It supports dynamic configuration for any number of databases and tables.
+func LoadConfig(logger *zap.Logger) (*model.Config, error) {
+	logger.Info("Loading configuration from environment variables")
+
+	gcpProjectID := getEnv(GCPProjectID, "")
+	bqDatasetID := getEnv(BQDatasetID, "")
+
+	if gcpProjectID == "" || bqDatasetID == "" {
+		return nil, fmt.Errorf("GCP_PROJECT_ID and BQ_DATASET_ID are required")
+	}
+
+	dbNames := getEnv(DatabasesKey, "")
+	if dbNames == "" {
+		return nil, fmt.Errorf("SYNC_DATABASES is required (comma-separated list of database identifiers)")
+	}
+
+	databases := make(map[string]*model.DatabaseConfig)
+	dbList := strings.Split(dbNames, ",")
+
+	for _, dbName := range dbList {
+		dbName = strings.TrimSpace(dbName)
+		if dbName == "" {
+			continue
+		}
+
+		dbConfig, err := loadDatabaseConfig(logger, dbName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load config for database '%s': %w", dbName, err)
+		}
+		databases[dbName] = dbConfig
+		logger.Info("Loaded database configuration",
+			zap.String("database", dbName),
+			zap.Int("tables", len(dbConfig.Tables)),
+		)
+	}
+
+	if len(databases) == 0 {
+		return nil, fmt.Errorf("no valid database configurations found")
+	}
+
+	maxOpen := parseInt(logger, DBMaxOpenConns, "10", 10)
+	maxIdle := parseInt(logger, DBMaxIdleConns, "10", 10)
+	defaultBatchSize := parseInt(logger, DefaultBatchSizeKey, "1000", 1000)
+	maxRowParseFailures := parseInt(logger, MaxRowParseFailuresKey, "100", 100)
+
+	syncTimeout := parseDuration(logger, SyncTimeoutKey, "10m", 10*time.Minute)
+	connMaxLifetime := parseDuration(logger, DBConnMaxLifetime, "1m", 1*time.Minute)
+
+	dryRun := parseBool(getEnv(DryRunKey, "false"))
+	createTables := parseBool(getEnv(CreateTablesKey, "true"))
+	truncateOnSync := parseBool(getEnv(TruncateOnSyncKey, "false"))
+
+	cfg := &model.Config{
+		GCPProjectID:        gcpProjectID,
+		BigQueryDatasetID:   bqDatasetID,
+		Databases:           databases,
+		SyncTimeout:         syncTimeout,
+		DateFormat:          getEnv(DateFormatKey, "2006-01-02T15:04:05Z07:00"),
```

Comment:
Shall we move this to the top to where the rest of the variables are defined?
```suggestion
		DateFormat:          dateFormat,
```